prov/efa/src/rdm/rxr_ep.h

```diff
@@ -279,6 +286,72 @@ static inline int rxr_need_sas_ordering(struct rxr_ep *ep)
 	return ep->msg_order & FI_ORDER_SAS;
 }
 
+static inline uint64_t rxr_get_host_id(char *host_id_file)
+{
+	FILE *fp = NULL;
+	char *host_id_str = NULL;
+	const size_t HOST_ID_LENGTH = 16;
+	long length = 0;
+	size_t read_length = 0;
+	uint64_t host_id = 0;
+
+	if (!host_id_file) {
+		EFA_WARN(FI_LOG_EP_CTRL, "Host id file is not specified\n");
+		goto out;
+	}
+
+	fp = fopen(host_id_file, "r");
+	if (!fp) {
+		EFA_WARN(FI_LOG_EP_CTRL, "Cannot open host id file: %s\n", host_id_file);
+		goto out;
+	}
+
+	for (char c = fgetc(fp); c != '\n' && c != EOF; c = fgetc(fp)) {
```

Comment:
I chatted with @lrbison about the general implementation. This particular length calc has been deprecated.

Overall I don't find `fscanf` suitable here, because we want to validate the string is a hex and has expected length. `fread` nicely returns the actual string length while `fscanf` does not.

Reply:
Why isn't `fscanf` suited here?

Reply:
With `fscanf`, I believe the intention is to read the hex into an integer directly, e.g.
```c
uint64_t id;

ret = fscanf(fp, "i-0%x", &id);
```

Consider the case
```
i-01234 // invalid input
```
vs
```
i-00000000000001234 // valid input
```

With `fscanf` we cannot tell apart those 2. On the other hand `fread` returns the number of bytes read from the file and we know the first input is invalid.